src/camera.py

```diff
@@ -0,0 +1,152 @@
+        return details
+
+    def _parse_camera_details(self) -> Optional[Dict[str, str]]:
+        """Parses the camera details and returns the camera info as a dictionary."""
+
+        if not self.details:
+            return None
+
+        details = self.details
+
+        city_start = details.find("City: ") + len("City: ")
+        city_end = details.find("Latitude:")
+        city = details[city_start:city_end]
+
+        region_start = details.find("Region:") + len("Region:")
+        region_end = details.find("City:")
+        region = details[region_start:region_end]
+
+        country_start = details.find("Country:") + len("Country:")
+        country_end = details.find("Country code:")
+        country = details[country_start:country_end]
+
+        country_code_start = details.find("Country code:") + len("Country code:")
+        country_code_end = details.find("Region:")
+        country_code = details[country_code_start:country_code_end]
+
+        return {
+            "city": city,
+            "region": region,
+            "country": country,
+            "country_code": country_code,
+        }
+
+    @exponential_backoff(
+        attempts=s.CAMERA_FETCH_ATTEMPTS,
+        initial_delay=s.INITIAL_RETRY_DELAY,
+        factor=s.RETRY_DELAY_FACTOR,
+        exception_types=(SaveImageException, FetchCamerasException),
+    )
+    def _save_image(self, image_file_path: str, camera_url: str, request_headers: Dict[str, str]) -> bool:
+        """Saves the image from the camera stream URL."""
+        try:
+            r = requests.get(camera_url, headers=request_headers, timeout=s.IMAGE_SAVE_TIMEOUT)
+            r.raise_for_status()
+
+            with open(image_file_path, "wb") as f:
+                f.write(r.content)
+            return True
+        except (RequestException, ReadTimeout) as e:
+            logger.error(f"error saving image: {e}")
+            raise SaveImageException(f"error saving image: {e}")
+
+    def _image_is_solid_color(self, image_file_path: str) -> bool:
+        """Checks if the image consists of a single color."""
+        image = cv2.imread(image_file_path)
+
+        if image is None:
+            logging.error("image is empty. skipping...")
+            return True
+
+        standard_deviation = std(image)
+
+        if standard_deviation == 0:
+            logging.info("image consists of a single color. skipping...")
+            return True
+
+        return False
+
+    def save_and_validate_image(self, image_file_path: str, request_headers: Dict[str, str]) -> bool:
+        """
+        Saves the image and validates that it is not a solid color.
+        Returns True if the image is saved and validated, False otherwise.
+        """
+        saved_successfully = self._save_image(image_file_path, self.stream_url, request_headers)
+
+        if saved_successfully:
+            if not self._image_is_solid_color(image_file_path=image_file_path):
+                return True
+            else:

```

Comment:
This `else` on 148 is probably unnecessary due to your return on the previous line which terminates the flow anyway. You could move the below else code up and outdent it as the path for any False.